### PR TITLE
Replaced earley by pacomb in the "metavar" example

### PR DESCRIPTION
**Pull request by craff**
~~Thanks to this PR bindlib and its examples compile under ocaml 5.0~~

### CODE REVIEW

File: benchmark/bench_de_bruijn.ml [hunk 1]
```diff
@@ -249,14 +249,14 @@ module Simple4 : Lambda = struct
     | Var of int
 
   type term = S.t * term_
-  
+
```
Comment (rlepigre) [inline benchmark/bench_de_bruijn.ml]: Changes to this file are unrelated to replacing earley by pacomp. I created an other MR that includes your changes plus some cleanup: https://github.com/rlepigre/ocaml-bindlib/pull/28. Please, review and merge. We can then rebase your change on top of that.

File: Makefile [hunk 2]
```diff
@@ -21,9 +21,13 @@ tests:
 .PHONY: tests
 
 bench: all
-	@ulimit -s unlimited && perf stat _build/default/benchmark/church.exe  > /dev/null 2>  perf.log
-	@ulimit -s unlimited && perf stat _build/default/benchmark/scott.exe   > /dev/null 2>> perf.log
-	@ulimit -s unlimited && perf stat _build/default/benchmark/lambdas.exe > /dev/null 2>> perf.log
+	@echo "to allow perf, run echo 0 > /proc/sys/kernel/perf_event_paranoid"
```
Comment (rlepigre) [inline Makefile]: This is too system-specific, I think. I would just revert the changes to this file (also not related to the MR).

**Review by rlepigre — COMMENTED** [review]
That looks good. I'm wondering: do we need to change the comment syntax in the example?

Also, could we keep the two parsers, and compile whichever depending on whether earley or pacomb is available? We could also add a menhir parser.

File: benchmark/bench_de_bruijn.ml (hunk 1 above)
Comment (craff) [reply benchmark/bench_de_bruijn.ml]: You are right, my emacs is removing all trailing space as soon as I edit the file and this gives bad commit with too many changes. We could have a sanity check for that in the workflow, as well as long lines?

File: Makefile (hunk 2 above)
Comment (craff) [reply Makefile]: To me is related to "make sure the example compiles" and it perf is not allowed you get a strange "255" error code. I wanted to make sure the user get a bit of information. I am replacing by something less specific now.

**Comment by craff** [comment]
> That looks good. I'm wondering: do we need to change the comment syntax in the example?

Just the pacomb does not have (yet) a predefined blank notion with opening/closing comments. It is always tricky
because of strings, so for an example I felt one line comments are simpler.
 
> Also, could we keep the two parsers, and compile whichever depending on whether earley or pacomb is available? We could also add a menhir parser.

If you know how to write a conditional opam file that does not force the user to install 3 parsing technologies. I still like more pacomb or earley because the parser is both readable and inside the file. By the way I wanted to make a compile time version of earley that would not allow dynamical grammar but that could be faster than camlyacc and menhir.

**Comment by craff** [comment]
Remark: we needed the lateest pacomb which is not on opam. I fixed that, but I think we need the metavar test to run. I am adding this.

File: benchmark/bench_de_bruijn.ml (hunk 1 above)
Comment (rlepigre) [reply benchmark/bench_de_bruijn.ml]: Yeah, I think I have code for that somewhere, I'll see if I can add it when I have a moment.

**Comment by rlepigre** [comment]
You'll need to rebase your MR on master, with something like `git fetch && git rebase origin/master`, and then force-push to the branch.

**Comment by craff** [comment]
bindlib and metatar tests work on 4.07, but with deps Timed >= 1.0, opam is not able to select Timed 1.0 on ocaml 4.07 and 1.1 on 4.08 and more recent. If you know how to fix that (or fix Timed 1.1), we can put back 4.07 support.

**Comment by craff** [comment]
All test run. I stop for now, you may review when you have time and try to recover 4.07, it is not nice to loose a version just because of Timed used in one example. Cheers.

**Comment by craff** [comment]
Hello Rodolphe ... I just did a PR on top of that one for issue #29. Do you think we should merge that one, or there is still some adjustment ?

**Comment by rlepigre** [comment]
I don't remember where this is at, I thought I was waiting for changes on your side. I'll have a closer look when I have a moment.

**Comment by craff** [comment]
I think the only thing to do was recover 4.07, and it seems to work now. I just needed to ask for dune 2.0, not 2.7, but it seems to work.

**Comment by craff** [comment]
On en est où avec cette PR...

**Comment by rlepigre** [comment]
> On en est où avec cette PR...

Aucune idée, j'ai perdu tout contexte. Je regarderai à nouveau quand j'ai un moment.

**Review by rlepigre — APPROVED** [review]
This looks good to me.
